lib/actions/map.js

```diff
@@ -74,15 +74,49 @@ export function setLocation(payload) {
 
 export function setLocationToCurrent(payload, intl) {
   return function (dispatch, getState) {
-    const currentPosition = getState().otp.location.currentPosition
+    const state = getState()
+    const currentPosition = state.otp.location.currentPosition
     if (currentPosition.error || !currentPosition.coords) return
-    payload.location = {
+
+    const location = {
       category: 'CURRENT_LOCATION',
       lat: currentPosition.coords.latitude,
       lon: currentPosition.coords.longitude,
       name: intl.formatMessage({ id: 'actions.map.currentLocation' })
     }
-    dispatch(settingLocation(payload))
+
+    // Immediately dispatch with current coordinates for responsiveness
+    dispatch(
+      settingLocation({
+        ...payload,
+        location
+      })
+    )
+
+    // Reverse geocode the current location to update the name
+    getGeocoder(state.otp.config.geocoder)
+      .reverse({ point: location })
+      .then((geocodedLocation) => {
+        if (geocodedLocation.name) {
+          // Create a new location object with the geocoded name but keep everything else the same
+          const updatedLocation = {
+            ...location,
+            name: geocodedLocation.name
+          }
+
+          // Update the location with the geocoded name
+          dispatch(
+            settingLocation({
```

Comment:
Instead of calling reverse geocoder in this function, just replace `settingLocation` with the `setLocation` call above and add `reverseGeocode: true` to the payload